### ISSUE TITLE
[Feature request] Switch tabs with keyboard shortcut

### ISSUE THREAD

**Issue by flof** [usebruno/bruno#736]
Bruno is great but I really miss the possibility to switch between tabs with the keyboard. 
I suggest to implement it like Firefox does it:

- Ctrl+PgUp: Switch to previous tab
- Ctrl+PgDown: Switch to next tab
- Ctrl+Tab: Switch to recently used tab i.e. pressed once opens the most recently used tab, pressed twice opens second least recently used tab. 

Especially the Ctrl+Tab shortcut I find very useful fir daily work.

**Comment by ParamjotSingh5** [comment]
@helloanoop  Can I be assigned this

**Comment by ParamjotSingh5** [comment]
Taking this up

**Comment by ParamjotSingh5** [comment]
Update: 
Got it working for `Ctrl+PgUp`: Switch to previous tab and `Ctrl+PgDown`: Switch to next tab

**Comment by flof** [comment]
Nice 👍

**Comment by ParamjotSingh5** [comment]
Pull request for feature: https://github.com/usebruno/bruno/pull/812

**Comment by flof** [comment]
@ParamjotSingh5 Move to prev/next tab works great.
But CTRL+TAB for switching to recently used tabs is not very useful how it is implemented. 

What is missing: 

Lets say i have multiple tabs open. Currently i am on tab A. Now i click on Tab B. Then i click on Tab C.
I want to move back to previous tab B, therefor press CTRL-TAB once. Now i want back to tab C. Since this is the most recently used tab, this should be possible by pressing CTRL-TAB once. But this doesn't work. Instead i get to tab A. 

This is different than how it is implemented in **Firefox** or **VSCode**.

I think the correct implementation would be:

- When user presses CTRL-TAB, don't switch tab yet, but instead count, how many times the user presses TAB again until releasing CTRL key
- Only when user releases CTRL key, move back in history so many steps as the user pressed TAB **while holding CTRL**. And important: also put the current tab to the stack, before switching.

Example: 

`CTRL_DOWN > TAB_PRESS > TAB_PRESS > CTRL_UP`: Put current tab to stack, then move back 2 steps in history

Among other benefits, this would allow to toggle between to 2 last recent tabs.

What do you think?

**Comment by ParamjotSingh5** [comment]
Sure @flof , I understand your concern, about _putting the current tab to the stack, before switching._  This can be done easily.
However, the other requirement that we have about counting the number of tab clicks and then moving back in history, this part seems a little tricky, considering the implementation of the keyboard event. I am looking into it. Will update accordingly.

Question: Do we have any use case within Bruno, where we are replicating this behavior of counting button clicks?

**Comment by flof** [comment]
No, i don't.

Maybe the CTRL+TAB behaviour needs more discussion.

In my opinion, the ultimate goal would be to implement CTRL+TAB like VSCode implemented it. But this is not trivial, since VSCode additionally shows a **popup menu** with the list of the most recent tabs, ordered by most recently opened, as soon as CTRL+TAB is pressed (and CTRL is hold down). The popup stays open as long as TAB is hold down. Initially, the first item is selected. With each press of TAB, the next item in the list is selected. On release of CTRL, the selected tab is opened and the popup closes.

My suggestion in my last post was to mimic this behavior, just **without** showing the popup menu, to keep the implementation more simple. But maybe the whole functionality isn't understandable for users without a popup menu. I'm not sure.

Firefox also show such a popup menu just like VSCode does, just with a different layout.

**Comment by ParamjotSingh5** [comment]
Yes, I am aware of this CTRL+TAB feature of VSCode. Last time, I was not sure, whether we were planning to implement the recent tabs history popup and, YES, the CTRL+TAB feature will be more understandable with the popup. 

We can move forward with this feature of replicating the CTRL+TAB feature from VSCode. I feel we can raise a separate feature request for this requirement.

What do you think?

**Comment by rmueller83** [comment]
Shouldn't this also include CTRL+<NUMBER>, e.g. CTRL+2 for accessing the second tab, like it is possible to do in web browsers?

**Comment by flof** [comment]
@ParamjotSingh5 You mean, do in this issue/PR only CTRL+PGUP and CTRL+PGDOWN and create a new issue/PR for CTRL+TAB?

**Comment by ParamjotSingh5** [comment]
> @ParamjotSingh5 You mean, do in this issue/PR only CTRL+PGUP and CTRL+PGDOWN and create a new issue/PR for CTRL+TAB?

@flof Yes, Keeping CTRL+TAB feature in a separate PR, as we are planning to have a popup window for tab access history. This can help to have limited and scoped changes per PR.

**Comment by flof** [comment]
@ParamjotSingh5 sounds like a good idea 👍

**Comment by ParamjotSingh5** [comment]
> @ParamjotSingh5 sounds like a good idea 👍

Cool, then I can alter the Pull request accordingly and will update here soon 🙂

**Comment by ParamjotSingh5** [comment]
Altered and pushed the changes into existing pull requests: https://github.com/usebruno/bruno/pull/812
We are limiting this feature request to CTRL+PGUP and CTRL+PGDN button functionality. Ideas related to CTRL+TAB button functionality will be taken up into a separate feature request. 

cc @flof

**Comment by ParamjotSingh5** [comment]
> Shouldn't this also include CTRL+_NUMBER_, e.g. CTRL+2 for accessing the second tab, like it is possible to do in web browsers?

We can register a feature request for this also, I believe it is a good-to-have feature.

**Comment by flof** [comment]
@ParamjotSingh5 looks good. Have approved, but i am not a maintainer, so there is still an approve from an maintainer required.

**Comment by ParamjotSingh5** [comment]
@helloanoop A Pull request corresponding to this feature is available for code review. https://github.com/usebruno/bruno/pull/812

**Comment by Jeremie-Chauvel** [comment]
Hi, interested in doing the ctrl-tab vscode/firefox like feature, I will look into it

**Comment by Jeremie-Chauvel** [comment]
Need some cleanup but I have what seems like a working solution:

https://github.com/usebruno/bruno/assets/28385941/e424a54b-3ce1-4792-8cb5-f2008b937ae0

 What do you think @flof, did I forget a edge case?

**Comment by flof** [comment]
Hi @Jeremie-Chauvel, the behavior looks good in the video. I tried run your feature branch locally but some trouble starting bruno. Is this the correct branch? https://github.com/Jeremie-Chauvel/bruno/tree/feature/ctrl-tab

Thanks for your effort!

**Comment by Jeremie-Chauvel** [comment]
Hi @flof,
I think I just pushed before going to sleep, not sure it was in a stable state. I will add Ctrl+maj+tab and let you know so you can test it 😊

**Comment by Jeremie-Chauvel** [comment]
Hi @flof ,
the pull request is ready should you want to test it: https://github.com/usebruno/bruno/pull/990

**Comment by flof** [comment]
Hi @Jeremie-Chauvel, i tested it locally and i have to say it works great. Good work!

**Comment by helloanoop** [comment]
Hey everyone!

Apologies for being late to the PRs and discussion around this.
So, I have merged #812 by @ParamjotSingh5 (which adds support for ctrl+pgup and ctrl+pgdown)

We are aligned with UX and the PR - #990 proposal by @Jeremie-Chauvel 
We are reviewing it. We should have have it merged in the next few days and it should be available in the upcoming `v1.28.0` release

**Comment by subiabre** [comment]
Since this is based off of Firefox can we also get the `Alt` tabs shortcut. Essentially, when you press `Alt` + `1` it switches to the first open tab, and so on.

**Comment by anusree-bruno** [comment]
Hi, the requested features are now available. Please check out the latest versions. For any additional shortcuts you’d like to see, please create new issues.
